geo/src/algorithm/intersects/rect.rs

```diff
@@ -40,7 +40,33 @@ where
 symmetric_intersects_impl!(Rect<T>, Point<T>);
 symmetric_intersects_impl!(Rect<T>, MultiPoint<T>);
 
-symmetric_intersects_impl!(Rect<T>, Polygon<T>);
+impl<T> Intersects<Polygon<T>> for Rect<T>
+where
+    T: GeoNum,
+{
+    fn intersects(&self, rhs: &Polygon<T>) -> bool {
+        /*
+        sufficient to show that any of these are true:
+        some corner of the polygon intersects the rectangle
+        some corner of the rectangle intersects the polygon
+        some edge of polygon (interior or exterior) intersects edge of rectangle
+        */
+
+        if has_disjoint_bboxes(self, rhs) {
+            return false;
+        }
+
+        // rhs.intersects(pt) is the most expensive, so check it last
```

Comment:
```suggestion
        // pt.intersects(polygon) is the most expensive, so check it last
```

This comment confused me for a minute because I didn't see where we're calling `rhs.intersects(pt)` anywhere, but I think this is what you mean.